`make all` fails with `Fatal error compiling: error: release version 17 not supported`

Running yamcs-quickstart with the `Makefile` and `docker` fails due to jdk 17 not being present on the maven docker image.

The docker image in the `Dockerfile` should be updated to use an image with jdk 17

---
```
Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-xml/3.0.0/plexus-xml-3.0.0.jar (93 kB at 391 kB/s)
2024-07-12 16:38:17 Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-compiler-javac/2.15.0/plexus-compiler-javac-2.15.0.jar (26 kB at 108 kB/s)
2024-07-12 16:38:17 Downloaded from central: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-utils/4.0.0/plexus-utils-4.0.0.jar (192 kB at 731 kB/s)
2024-07-12 16:38:17 [INFO] Recompiling the module because of changed dependency.
2024-07-12 16:38:17 [INFO] Compiling 2 source files with javac [debug release 17] to target/classes
2024-07-12 16:38:17 [INFO] ------------------------------------------------------------------------
2024-07-12 16:38:17 [INFO] BUILD FAILURE
2024-07-12 16:38:17 [INFO] ------------------------------------------------------------------------
2024-07-12 16:38:17 [INFO] Total time:  18.358 s
2024-07-12 16:38:17 [INFO] Finished at: 2024-07-12T23:38:17Z
2024-07-12 16:38:17 [INFO] ------------------------------------------------------------------------
2024-07-12 16:38:17 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.13.0:compile (default-compile) on project myproject: Fatal error compiling: error: release version 17 not supported -> [Help 1]
2024-07-12 16:38:17 [ERROR] 
2024-07-12 16:38:17 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
2024-07-12 16:38:17 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
2024-07-12 16:38:17 [ERROR] 
2024-07-12 16:38:17 [ERROR] For more information about the errors and possible solutions, please read the following articles:
2024-07-12 16:38:17 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -1,3 +1,3 @@
-FROM maven:3.6-openjdk-11
+FROM maven:3.8.5-openjdk-17
 
 WORKDIR /yamcs

--- a/docker/Makefile
+++ b/docker/Makefile
@@ -10,17 +10,17 @@ all: ## run all, yamcs-up (yamcs-down) and yamcs-simulator
 	$(MAKE) yamcs-simulator
   
 yamcs-up: | yamcs-down ## bring up yamcs system
-	docker-compose up -d
+	docker compose up -d
 
 yamcs-down: ## bring down yamcs system
-	docker-compose down -v --remove-orphans
+	docker compose down -v --remove-orphans
 
 yamcs-simulator: yamcs-up ## run yamcs simulator
 	@echo "connect via http://localhost:8090/ system make take about 50 seconds to startup" && \
 	cd .. && $(PYTHON) ./simulator.py
 
 yamcs-shell: ## shell into yamcs container
-	docker-compose up -d && docker run -it yamcs "bash"
+	docker compose up -d && docker compose exec yamcs bash
 
 help:
 	@printf "\033[37m%-30s\033[0m %s\n" "#----------------------------------------------------------------------------------"

